### ISSUE TITLE
reload occasionnally fails on windows

### ISSUE THREAD

**Issue by OlegYch** [sbt/sbt#6471]
## steps
sbt version from somewhere around 1.3.10 to 1.5.0
Windows OS
`reload`

## problem

```
 reload                                                                  
[info] welcome to sbt 1.5.0 (Oracle Corporation Java 1.8.0_281)          
[info] loading settings for project global-plugins from plugins.sbt ...  
[info] loading global plugins from C:\Users\OlegYch\.sbt\1.0\plugins     
[error] stack trace is suppressed; run last Compile / unmanagedSources for the full output
[error] (Compile / unmanagedSources / allInputPathsAndAttributes) java.lang.IllegalArgumentException: 'other' has different root
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
```
reloading again usually works

**Comment by eed3si9n** [comment]
@OlegYch Thanks for the report. Next time it happens could you try `l` and display the stacktrace plz?

**Comment by OlegYch** [comment]
```
 reload
[info] welcome to sbt 1.5.0 (Oracle Corporation Java 1.8.0_281)
[info] loading settings for project global-plugins from plugins.sbt ...
[info] loading global plugins from C:\Users\OlegYch\.sbt\1.0\plugins
[error] stack trace is suppressed; run last Compile / unmanagedSources for the full output
[error] (Compile / unmanagedSources / allInputPathsAndAttributes) java.lang.IllegalArgumentException: 'other' has different root
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
[debug] > Exec(reload, Some(33be7c7f-a864-40f8-97f2-16e36641a41d), Some(CommandSource(console0)))
[debug] > Exec(sbtStashOnFailure, None, Some(CommandSource(console0)))
[debug] > Exec(onFailure loadFailed, None, Some(CommandSource(console0)))
[debug] > Exec(loadp, None, Some(CommandSource(console0)))
[info] welcome to sbt 1.5.0 (Oracle Corporation Java 1.8.0_281)
[debug]       Load.loadUnit: plugins took 21.972799ms
[debug]       Load.loadUnit: defsScala took 0.0047ms
[debug] [Loading] Scanning directory C:\Users\OlegYch\.sbt\1.0\plugins
[debug]         Load.loadUnit: mkEval took 0.1562ms
[debug] [Loading] Found non-root projects
[debug] [Loading] Done in C:\Users\OlegYch\.sbt\1.0\plugins, returning: ()
[debug] deducing auto plugins based on known facts Set(Atom(sbt.plugins.CorePlugin)) and clauses Clauses(Clause(Atom(sbt.plugins.JvmPlugin),Set(Atom(sbt.plugins.IvyPlugin)))
[debug] Clause(Atom(sbt.ScriptedPlugin),Set(Atom(sbt.plugins.JvmPlugin)))
[debug] Clause(Atom(sbt.plugins.SbtPlugin),Set(Atom(sbt.ScriptedPlugin)))
[debug] Clause(Atom(sbt.plugins.SemanticdbPlugin),Set(Atom(sbt.plugins.JvmPlugin)))
[debug] Clause(Atom(sbt.plugins.JUnitXmlReportPlugin),Set(Atom(sbt.plugins.JvmPlugin)))
[debug] Clause(Atom(sbt.plugins.MiniDependencyTreePlugin),Set(Atom(sbt.plugins.JvmPlugin)))
[debug] Clause(Atom(sbt.plugins.CorePlugin),Set(Atom(sbt.plugins.IvyPlugin)))
[debug] Clause(Atom(sbt.plugins.IvyPlugin),Set(Atom(sbt.plugins.JvmPlugin)))
[debug] Clause(Atom(sbt.plugins.JvmPlugin),Set(Atom(sbt.plugins.SemanticdbPlugin)))
[debug] Clause(Atom(sbt.plugins.JvmPlugin),Set(Atom(sbt.plugins.JUnitXmlReportPlugin)))
[debug] Clause(Atom(sbt.plugins.CorePlugin),Set(Atom(sbt.plugins.Giter8TemplatePlugin)))
[debug] Clause(Atom(sbt.plugins.JvmPlugin),Set(Atom(sbt.plugins.MiniDependencyTreePlugin))))
[debug]   :: deduced result: Matched(sbt.plugins.CorePlugin,sbt.plugins.Giter8TemplatePlugin,sbt.plugins.IvyPlugin,sbt.plugins.JvmPlugin,sbt.plugins.MiniDependencyTreePlugin,sbt.plugins.JUnitXmlReportPlugin,sbt.plugins.SemanticdbPlugin) [debug] Plugins.deducer#function took 1.3829 ms
[info] loading settings for project global-plugins from plugins.sbt ...
[debug]           Load.resolveProject(global-plugins) took 3.0434ms
[debug]         Load.loadTransitive: finalizeProject(Project(id global-plugins, base: C:\Users\OlegYch\.sbt\1.0\plugins, plugins: List(<none>))) took 4.938ms
[debug] [Loading] Done in C:\Users\OlegYch\.sbt\1.0\plugins, returning: (global-plugins)
[debug]       Load.loadUnit: loadedProjectsRaw took 16.117901ms
[debug]       Load.loadUnit: cleanEvalClasses took 0.5631ms
[debug]     Load.loadUnit(file:/C:/Users/OlegYch/.sbt/1.0/plugins/, ...) took 40.3838ms
[debug]   Load.apply: load took 41.436001ms
[debug]   Load.apply: resolveProjects took 0.075ms
[debug]   Load.apply: finalTransforms took 5.491001ms
[debug]   Load.apply: config.delegates took 0.3133ms
[debug]   Load.apply: Def.make(settings)... took 43.1272ms
[debug]   Load.apply: structureIndex took 31.425ms
[debug]   Load.apply: mkStreams took 0.002201ms
[info] loading global plugins from C:\Users\OlegYch\.sbt\1.0\plugins
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
[error] java.lang.IllegalArgumentException: 'other' has different root
[error]         at sun.nio.fs.WindowsPath.relativize(Unknown Source)
[error]         at sun.nio.fs.WindowsPath.relativize(Unknown Source)
[error]         at com.swoval.files.CachedDirectoryImpl.update(CachedDirectoryImpl.java:245)
[error]         at com.swoval.files.NioPathWatcher.update(NioPathWatcher.java:354)
[error]         at com.swoval.files.NioPathWatcher.register(NioPathWatcher.java:201)
[error]         at com.swoval.files.FileCacheDirectoryTree.register(FileCacheDirectoryTree.java:417)
[error]         at com.swoval.files.FileCachePathWatcher.register(FileCachePathWatcher.java:24)
[error]         at com.swoval.files.FileTreeRepositoryImpl.register(FileTreeRepositoryImpl.java:101)
[error]         at sbt.internal.nio.FileTreeRepositoryImpl.register(FileTreeRepositoryImpl.scala:130)
[error]         at sbt.nio.Settings$.$anonfun$inputPathSettings$3(Settings.scala:157)
[error]         at sbt.nio.Settings$.$anonfun$inputPathSettings$3$adapted(Settings.scala:157)
[error]         at scala.collection.immutable.List.foreach(List.scala:431)
[error]         at sbt.nio.Settings$.$anonfun$inputPathSettings$2(Settings.scala:157)
[error]         at sbt.nio.Settings$.$anonfun$inputPathSettings$2$adapted(Settings.scala:156)
[error]         at scala.Option.foreach(Option.scala:407)
[error]         at sbt.nio.Settings$.$anonfun$inputPathSettings$1(Settings.scala:156)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.util.concurrent.FutureTask.run(Unknown Source)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[error]         at java.util.concurrent.FutureTask.run(Unknown Source)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[error]         at java.lang.Thread.run(Unknown Source)
[error] (Compile / unmanagedSources / allInputPathsAndAttributes) java.lang.IllegalArgumentException: 'other' has different root
[debug] > Exec(loadFailed, None, Some(CommandSource(console0)))
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
[debug] > Exec(last, None, Some(CommandSource(console0)))
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
```

**Comment by OlegYch** [comment]
still an issue on 1.5.5

**Comment by eed3si9n** [comment]
Looking at the stacktrace it seems like FileTreeRepository's `register` is failing due to `WindowsPath.relativize`.
Could you report this also to swoval/swoval please?

**Comment by OlegYch** [comment]
this only happens if there are *.sbt files in ~/.sbt/1.0/plugins

**Comment by eatkins** [comment]
I feel like technically we shouldn't close issues (this was autoclosed by my merging https://github.com/sbt/io/pull/325) until there is a binary released but this is fixed now in the latest io code.

**Comment by OlegYch** [comment]
looks fixed in 1.6
